packages/plugin-core/src/ExtensionProvider.ts

```diff
@@ -0,0 +1,21 @@
+import { IDendronExtension } from "./dendronExtensionInterface";
+import { DendronError } from "@dendronhq/common-all";
+import _ from "lodash";
+
+export class ExtensionProvider {
```

Comment:
The idea with this class is that we can still query out statically to grab the extension (especially during transition) without having to depend on `workspace` file or concrete implementation of extension.

Reply:
Sounds good - can you add a comment to this class w/ this explanation? Also - make a note there that any _new_ classes should **not** use these statics and instead get an instance of `IDendronExtension` passed into their constructors

Reply:
Added explanation, and @deprecated getExtension() in workspace.ts (along with getDWorkspace(),  getEngine()) with a comment to go their mirroring ExtensionProvider methods.